jax_distribution/systems/ff_ippo_rware.py

```diff
@@ -317,15 +319,133 @@ def learner_fn(runner_state: RunnerState) -> ExperimentOutput:
             _update_step, in_axes=(0, None), axis_name="batch"
         )
 
-        runner_state, metric = jax.lax.scan(
-            batched_update_step, runner_state, None, config["NUM_UPDATES"]
+        runner_state, (metric, loss_info) = jax.lax.scan(
+            batched_update_step, runner_state, None, config["NUM_UPDATES_PER_EVAL"]
         )
-        return {"runner_state": runner_state, "metrics": metric}
+        total_loss, (value_loss, loss_actor, entropy) = loss_info
+        return {
+            "runner_state": runner_state,
+            "metrics": metric,
+            "total_loss": total_loss,
+            "value_loss": value_loss,
+            "loss_actor": loss_actor,
+            "entropy": entropy,
+        }
 
     return learner_fn
 
 
-def learner_setup(env: Environment, config: Dict) -> Tuple[callable, RunnerState]:
+def get_evaluator_fn(env: Environment, apply_fn: callable, config: dict) -> callable:
+    """Get the evaluator function."""
+
+    def eval_one_episode(params, runner_state) -> Tuple:
+        """Evaluate one episode. It is vectorized over the number of evaluation episodes."""
+
+        def _env_step(runner_state: Tuple) -> Tuple:
+            """Step the environment."""
+            # PRNG keys.
+            rng, env_state, last_timestep, step_count_, return_ = runner_state
```

Comment:
Why do step_count_ and return_ have an _ at the end?

Reply:
yeah this is the difference between this step and the old function, this one we can directly cumulative the val of step count and return therefore they are a part of the runner_state

Reply:
I will update the name of runner_state to eval_state and make it a named tuple